src/interaction/PriorityObservable.jl

```diff
@@ -0,0 +1,142 @@
+
+
+mutable struct PriorityObservable{T} <: AbstractObservable{T}
+    listeners::Vector{Pair{Int8, Vector{Any}}}
+    val::T
+
+    PriorityObservable{T}() where {T} = new{T}(Pair{Int8, Vector{Any}}[])
+    PriorityObservable{T}(val) where {T} = new{T}(Pair{Int8, Vector{Any}}[], val)
+    # Construct an Observable{Any} without runtime dispatch
+    PriorityObservable{Any}(@nospecialize(val)) = new{Any}(Pair{Int8, Vector{Any}}[], val)
+end
+
+"""
+    PriorityObservable(value)
+
+Creates a new `PriorityObservable` holding the given `value`.
+
+A `po = PriorityObservable` differs from a normal `Observable` (or `Node`) in 
+two ways:
+1. When registering a function to `po` you can also give it a `priority`.
+Functions with higher priority will be evaluated first.
+2. Each registered function is assumed to return a `Bool`. If `true` is returned
+the observable will stop calling the remaining observer functions.
+
+In the following example we have 3 listeners to a `PriorityObservable`. When 
+triggering the `PriorityObservable` you will first see `"First"` as it is the 
+callback with the highest priority. After that `"Second"` will be printed and 
+stop further execution. `"Third"` will therefore not be printed.
+```
+po = PriorityObservable(1)
+on(x -> begin printstyled("First\n", color=:green); false end, po, priority = 1)
+on(x -> begin printstyled("Second\n", color=:orange); true end, po, priority = 0)
+on(x -> begin printstyled("Third\n", color=:red); false end, po, priority = -1)
+po[] = 2
+```
+
+Note that `PriorityObservable` does not implement `map`. If you wish to know
+whether any observer function returned `true`, you can check the output of
+`setindex!(po, val)`.
+"""
+PriorityObservable(val::T) where {T} = PriorityObservable{T}(val)
+
+Base.getindex(observable::PriorityObservable) = observable.val
+function Base.setindex!(observable::PriorityObservable, val; notify=(x)->true)
+    observable.val = val
+    for f in listeners(observable)
+        if notify(f)
+            output = if f isa InternalFunction
+                f(val)
+            else
+                Base.invokelatest(f, val)
+            end
+            output && return true
+        end
+    end
+    return false
+end
+
+# reverse order so that the highest priority is notified first
+listeners(o::PriorityObservable) = (f for p in reverse(o.listeners) for f in p[2])
+
+function on(@nospecialize(f), observable::PriorityObservable; weak::Bool = false, priority = Int8(0))
+    if Core.Compiler.return_type(f, (typeof(observable.val),)) !== Bool
+        sanitized_func = x -> begin f(x); false end
+        @warn(
+            "Observer functions of PriorityObservables must return a Bool to " *
+            "specify whether the update is consumed (true) or should " *
+            "propagate (false) to other observer functions. The given " *
+            "function has been wrapped to always return false."
+        )
+    else
+        sanitized_func = f
+    end
```

Comment:
This should allow the old syntax to work, but with a warning

Reply:
I also added a `show_backtrace(...)` here now, so it's easier to find what's causing the warning. Is it possible to attach that to a warning directly?